src/jedi_bundle/utils/git.py

```diff
@@ -66,29 +67,34 @@ def repo_is_reachable(logger, url, username, token):
 # --------------------------------------------------------------------------------------------------
 
 
-def repo_has_branch(logger, url, branch, is_tag=False):
+def repo_has_branch(logger, url, branch, is_tag=False, is_commit=False):
 
-    # Command to check if branch exists and pass exit code back
-    heads_or_tags = '--heads'
-    if is_tag:
-        heads_or_tags = '--tags'
+    if is_commit:
+        # Cannot check for a commit before cloning, check after cloning
+        return True
```

Comment:
If useful, you can actually use the [GitHub API](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#get-a-commit) to check this.

```python
import requests

# A commit that exists
r = requests.get(
    "https://api.github.com/repos/geos-esm/swell/commits/1c77b32592d15a8f83f224549cc81c5624557d3b",
    headers = {
        "Accept": "application/vnd.github+json",
        "X-GitHub-Api-Version": "2022-11-28"
    }
)
r.ok  # Returns True
r.status_code  # Returns 200

# A commit that doesn't exist
r2 = requests.get(
    "https://api.github.com/repos/geos-esm/swell/commits/12345678abcdef",
    headers = {
        "Accept": "application/vnd.github+json",
        "X-GitHub-Api-Version": "2022-11-28"
    }
)
r2.ok  # Returns False
r2.status_code  # Returns 422 (one of the error codes)
```

Note: `requests` is a 3rd party library, albeit a very common one, so if we don't use it already, we may have to add it to our `requirements.txt`. You can also use `urllib` from the standard library, but it's a bit more code.

Note also: GitHub rate-limits unauthenticated API requests to 60 per hour. Depending on exactly how frequently this runs, that should be OK, but if we start hitting rate-limiting errors, we may need to set up some GitHub OAuth tokens.

Reply:
Awesome! I'll give this a try

Reply:
Funny enough, `requests` is already used in the `repo_is_reachable` function!